src/include/hip/rpp_hip_common.hpp

```diff
@@ -1429,6 +1466,81 @@ __device__ __forceinline__ void rpp_hip_layouttoggle24_pln3_to_pkd3(T *pixpln3Pt
     *pixpln3Ptr_T24 = pixpkd3_T24;
 }
 
+// ------------------------- Set 8 - Loads to uint / int --------------------------
+
+__device__ __forceinline__ void rpp_hip_load8_to_uint8(uchar *srcPtr, d_uint8 *srcPtr_ui8)
+{
+    d_uchar8 src_uc8;
+    *(d_uchar8_s *)&src_uc8 = *(d_uchar8_s *)srcPtr;
+
+    srcPtr_ui8->ui4[0] = rpp_hip_pack_uint4(src_uc8.uc4[0]);
+    srcPtr_ui8->ui4[1] = rpp_hip_pack_uint4(src_uc8.uc4[1]);
+}
+
+__device__ __forceinline__ void rpp_hip_load8_to_int8(schar *srcPtr, d_int8 *srcPtr_i8)
+{
+    d_schar8_s src_sc8;
+    *(d_schar8_s *)&src_sc8 = *(d_schar8_s *)srcPtr;
+
+    rpp_hip_pack_int8(&src_sc8, srcPtr_i8);
+}
+
+__device__ __forceinline__ void rpp_hip_load24_pln3_to_uint24_pln3(uchar *srcPtr, uint increment, d_uint24 *srcPtr_ui24)
+{
+    d_uchar24 src_uc24;
```

Comment:
I would discourage the use of this custom datatypes except for few cases where it makes the code.
In this case I would suggest accessing source directly instead of copying the data into src_uc24.
Just use ptrR, pG and pB (cast it to uchar4)

Reply:
As we discussed all the custom datatypes beyond 24 have all been removed. 

The only reason to retain 24 is the ability to do a one-shot vectorized global memory access - read/write of (RGB pixel * vector of 8 pixels = 24 elements) in a struct. Removing this would need re-evaluation of all loads and stores since they are common to multiple functions. We could run experiments for this, and issue a common PR if load/store funcs need any mods, since this could result in a higher number of global memory accesses causing performance changes.